File: D4ST-Api/Models/Interfaces/IMainStatAffix.cs
using D4ST_Api.Models.Enums;

namespace D4ST_Api.Models.Interfaces
{
    public interface IMainStatAffix
    {
        int BasicAmount { get; set; }
        int BonusAmount { get; set; }
        int BonusAmountPercentage { get; set; }
        int TotalAmount { get; set; }
        decimal BasicRegen { get; set; }
        decimal BonusRegen { get; set; }
        decimal BonusRegenPercentage { get; set; }
        decimal TotalRegen { get; set; }
        int CalculateAmount(IClassDefinition classStats);
        decimal CalculateRegen(IClassDefinition classStats);
    }
}

File: D4St-Spa/src/Models/DTOs/IMainStatAffix.ts
import { IClassDefinition } from './IClassDefinition';

export interface IMainStatAffix {
    BasicAmount: number;
    BonusAmount: number;
    BonusAmountPercentage: number;
    BasicRegen: number;
    BonusRegen: number;
    BonusRegenPercentage: number;
    CalculateAmount(classStats: IClassDefinition);
    CalculateRegen(classStats: IClassDefinition);
}
